app/code/community/Hps/Securesubmit/controllers/PaypalController.php

```diff
@@ -109,15 +102,30 @@ public function startAction()
                 $this->getResponse()->setRedirect($url);
                 return;
             }
+            $result = array();
         } catch (Mage_Core_Exception $e) {
             $this->_getCheckoutSession()->addError($e->getMessage());
             Mage::log(Mage::helper('hps_securesubmit')->__("Error creating PayPal session: %s", $e->getMessage()), Zend_Log::WARN);
+            $result = array(
+                'result' => 'error',
+                'message' => $e->getMessage(),
+                'redirect' => Mage::getUrl('checkout/cart'),
+            );
         } catch (Exception $e) {
             $this->_getCheckoutSession()->addError($this->__('Unable to start PayPal Checkout.'));
+            $this->_getCheckoutSession()->addError($e->getMessage());
```

Comment:
Do we need this here? I don't think the exception message will always be consumer friendly.

Reply:
Yes it will be better to have this message. 'Unable to start PayPal Checkout.' is more generic, users will not understand what went wrong.